### PR TITLE
Jl/search filter panel

### PR DESCRIPTION
**Pull request by neroliang**
This change is [Reviewable](https://reviewable.io/reviews/digitalnz/supplejack_website/24)

### CODE REVIEW

**Comment by Reinkaos** [comment]
Review status: 0 of 11 files reviewed at latest revision, 30 unresolved discussions, some commit checks failed.

---

*[app/assets/javascripts/records.js, line 35 \[r3\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KEj3EDTVRZPh3iGB6nZ:-KEj3EDTVRZPh3iGB6n_:2097897961) ([raw file](https://github.com/digitalnz/supplejack_website/blob/d1a103c37e4aab01f92a53ae8f4354129edcdb65/app/assets/javascripts/records.js#L35)):*
Remove

---

*[app/assets/javascripts/records.js, line 37 \[r3\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KEj3FMMWxg5oZPrUShR:-KEj3FMMWxg5oZPrUShS:2097897961) ([raw file](https://github.com/digitalnz/supplejack_website/blob/d1a103c37e4aab01f92a53ae8f4354129edcdb65/app/assets/javascripts/records.js#L37)):*
Remove

---

*[app/assets/javascripts/records.js, line 52 \[r3\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KEj3CZa17RVuwVTdmXh:-KEj3CZa17RVuwVTdmXi:2097897961) ([raw file](https://github.com/digitalnz/supplejack_website/blob/d1a103c37e4aab01f92a53ae8f4354129edcdb65/app/assets/javascripts/records.js#L52)):*
Remove

---

*[app/assets/javascripts/records.js, line 53 \[r3\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KEj3DHX4Np0q_9hZBtd:-KEj3DHX4Np0q_9hZBte:2097897961) ([raw file](https://github.com/digitalnz/supplejack_website/blob/d1a103c37e4aab01f92a53ae8f4354129edcdb65/app/assets/javascripts/records.js#L53)):*
Remove

---

*[app/assets/javascripts/records.js, line 60 \[r3\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KEj3Bem0zSkI5usZ-4b:-KEj3Bem0zSkI5usZ-4c:2097897961) ([raw file](https://github.com/digitalnz/supplejack_website/blob/d1a103c37e4aab01f92a53ae8f4354129edcdb65/app/assets/javascripts/records.js#L60)):*
Remove

---

*[app/assets/javascripts/records.js, line 93 \[r3\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KEj38neASan_-Cs82yC:-KEj38neASan_-Cs82yD:2097897961) ([raw file](https://github.com/digitalnz/supplejack_website/blob/d1a103c37e4aab01f92a53ae8f4354129edcdb65/app/assets/javascripts/records.js#L93)):*
Remove

---

*[app/assets/javascripts/records.js, line 94 \[r3\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KEj39PJdxYYCSROLyW3:-KEj39PJdxYYCSROLyW4:2097897961) ([raw file](https://github.com/digitalnz/supplejack_website/blob/d1a103c37e4aab01f92a53ae8f4354129edcdb65/app/assets/javascripts/records.js#L94)):*
Remove

---

*[app/assets/javascripts/records.js, line 186 \[r3\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KEj35KkdXVvNpFUqUwz:-KEj35KkdXVvNpFUqUx-:2097897961) ([raw file](https://github.com/digitalnz/supplejack_website/blob/d1a103c37e4aab01f92a53ae8f4354129edcdb65/app/assets/javascripts/records.js#L186)):*
Remove

---

*[app/assets/javascripts/records.js, line 190 \[r3\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KEj365pr2pMtrIxwa4j:-KEj365pr2pMtrIxwa4k:2097897961) ([raw file](https://github.com/digitalnz/supplejack_website/blob/d1a103c37e4aab01f92a53ae8f4354129edcdb65/app/assets/javascripts/records.js#L190)):*
Remove

---

*[app/assets/javascripts/records.js, line 271 \[r3\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KEj32Od129Beil9jrh3:-KEj32Od129Beil9jrh4:2097897961) ([raw file](https://github.com/digitalnz/supplejack_website/blob/d1a103c37e4aab01f92a53ae8f4354129edcdb65/app/assets/javascripts/records.js#L271)):*
Remove

---

*[app/assets/javascripts/records.js, line 282 \[r3\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KEj319Pp-NcSmVjawDD:-KEj319Pp-NcSmVjawDE:2097897961) ([raw file](https://github.com/digitalnz/supplejack_website/blob/d1a103c37e4aab01f92a53ae8f4354129edcdb65/app/assets/javascripts/records.js#L282)):*
Remove

---

*[app/assets/javascripts/records.js, line 285 \[r3\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KEj30Asd-YqsVQc7UxU:-KEj30Asd-YqsVQc7UxV:2097897961) ([raw file](https://github.com/digitalnz/supplejack_website/blob/d1a103c37e4aab01f92a53ae8f4354129edcdb65/app/assets/javascripts/records.js#L285)):*
Remove

---

*[app/assets/javascripts/records.js, line 299 \[r3\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KEj3--QHxsg4-dkmTx9:-KEj3--QHxsg4-dkmTxA:610327397) ([raw file](https://github.com/digitalnz/supplejack_website/blob/d1a103c37e4aab01f92a53ae8f4354129edcdb65/app/assets/javascripts/records.js#L299)):*
Remove.

---

*[app/views/records/_search_tabs.html.haml, line 16 \[r3\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KEj2iDSCh2TWtO7vXj4:-KEj2iDSCh2TWtO7vXj5:552312753) ([raw file](https://github.com/digitalnz/supplejack_website/blob/d1a103c37e4aab01f92a53ae8f4354129edcdb65/app/views/records/_search_tabs.html.haml#L16)):*
Use https://api.jquery.com/event.preventdefault/ instead of inline JS :). 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24)*

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-1.9.3-p484
+2.3.0-dev
```
Comment (UberMouse) [inline .ruby-version]: This should be 2.3.0 (with out "-dev")

File: app/views/records/_filters.html.haml [hunk 2]
```diff
@@ -1,8 +1,9 @@
 %body
   %div.clearfix
-    %button.filter-btn.button.menu{value: 0} Filters
+    %a{:href => '#', "value" => 0, "data-dropdown" =>"search_filter", :class =>"button filter-btn button menu"}Filters
```
Comment (UberMouse) [inline app/views/records/_filters.html.haml]: Hash rockets! Please replace with non hash-rockets

File: app/views/records/_filters.html.haml [hunk 3]
```diff
@@ -1,8 +1,9 @@
...
+
     = render 'active_filters'
-  .filter-container.menu.clearfix{style: "display:none", tabindex: "-1"}
+  #search_filter.filter-container.menu.content{"data-dropdown-content" => ""}
```
Comment (UberMouse) [inline app/views/records/_filters.html.haml]: Hash rockets! Please replace with non hash-rockets

File: app/views/shared/_search.html.haml [hunk 4]
```diff
@@ -6,14 +6,19 @@
   \#
   \# Supplejack was created by DigitalNZ at the National Library of NZ and the Department of Internal Affairs.
   \# http://digitalnz.org/supplejack
-= form_tag records_path, :method => :get, class: 'search clearfix' do
+= form_tag records_path, :method => :get, class: 'search clearfix', id: 'search-form' do
```
Comment (UberMouse) [inline app/views/shared/_search.html.haml]: Hash rockets! Please replace with non hash-rockets

**Comment by UberMouse** [comment]
Reviewed 7 of 9 files at r1, 1 of 2 files at r2, 1 of 3 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 35 unresolved discussions, some commit checks failed.

---

*[.ruby-version, line 1 \[r4\]](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24#-KF1D9H4MEgWppleMftv:-KF1D9H5j7qWfTFVsh9X:1049844869) ([raw file](https://github.com/digitalnz/supplejack_website/blob/781ad89982ede6ff6cff4aea2bbc24fec3ff8471/.ruby-version#L1)):*
This should be 2.3.0 (without "-dev")

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/digitalnz/supplejack_website/24)*

**Pull request merged**